File: scripts/stock_filter.py
"""Exclude cash / placeholder rows from holdings, diffs, and signals."""

from __future__ import annotations

import re

CASH_NAME_RE = re.compile(r"설정현금|현금및|예금|CASH|머니마켓|MMF|원화현금|원화$", re.I)


def is_trackable_stock(stock_code: str | None, stock_name: str | None = None) -> bool:
    code = (stock_code or "").strip()
    if len(code) != 6 or not code.isdigit():
        return False
    if code == "000000":
        return False
    name = (stock_name or "").strip()
    if name and CASH_NAME_RE.search(name):
        return False
    return True

File: lib/stock-filter.ts
/** Exclude cash / placeholder rows from stock rankings and signals. */

import { isListedEquity } from "@/lib/equity-classify";

const CASH_NAME_PATTERN = /설정현금|현금및|예금|CASH|머니마켓|MMF|원화현금|원화$/i;

export function isTrackableStock(
  stockCode: string | null | undefined,
  stockName?: string | null,
): boolean {
  const code = (stockCode ?? "").trim();
  if (!/^[0-9]{6}$/.test(code)) return false;
  if (code === "000000") return false;

  const name = (stockName ?? "").trim();
  if (name && CASH_NAME_PATTERN.test(name)) return false;

  return true;
}

/** 주식(국내·해외)만 — 채권·CP·TRS·스왑·ETF·선물 등 차트 없는 자산 제외 */
export function isEquityStock(
  stockCode: string | null | undefined,
  stockName?: string | null,
): boolean {
  return isListedEquity(stockName, stockCode, "all");
}
